src/Http/Controllers/PageController.php

```diff
@@ -13,6 +13,23 @@ class PageController extends BaseController
 {
     public function __invoke(Request $request, Page $page, ?Page $childPage = null, Page $grandChildPage = null)
     {
+        // If location based grand child is only child, redirect!!
+        if ($grandChildPage) {
+            if ($grandChildPage->location_based && $grandChildPage->is_only_child) {
+                return redirect(url($grandChildPage->path));
+            }
```

Comment:
I guess in this context you might be using it as a location-based page. There can't be a location under a location. Only a location (child page) under a Marker (page/parent page).